gui/src/ts/components/benchmark/BenchmarkAnswerPage.tsx

```diff
@@ -1,5 +1,222 @@
+import * as _ from "lodash";
+import * as ReactLoader from "react-loader";
+import {ApolloException} from "@tetherless-world/twxplore-base";
+import {FatalErrorModal} from "components/error/FatalErrorModal";
+import {
+  Grid,
+  Breadcrumbs,
+  Typography,
+  Card,
+  CardContent,
+} from "@material-ui/core";
+import {Hrefs} from "Hrefs";
 
-export const BenchmarkAnswerPage: React.FunctionComponent = () => (
-  <div>Benchmark answer page</div>
+//localhost:9001/benchmark/benchmark0/dataset/benchmark0-test/submission/benchmark0-submission/question/benchmark0-test-0
+
+interface BenchmarkPathParams {
+  benchmarkId: string;
+  datasetId: string;
+  submissionId: string;
+  questionId: string;
+}
+
+const BenchmarkBreadcrumbs: React.FunctionComponent<Partial<
+  BenchmarkPathParams
+>> = ({benchmarkId, datasetId, submissionId, questionId}) => {
+  const breadcrumbsChildren: React.ReactNode = (() => {
+    const breadcrumbs: React.ReactNodeArray = [
+      <Link to={Hrefs.benchmarks}>Benchmarks</Link>,
+    ];
+
+    if (!benchmarkId) {
+      return breadcrumbs;
+    }
+
+    const benchmark = Hrefs.benchmark({id: benchmarkId});
+    breadcrumbs.push(<Link to={benchmark.home}>{benchmarkId}</Link>);
+
+    if (!datasetId) {
+      return breadcrumbs;
+    }
+
+    breadcrumbs.push(<Typography>Datasets</Typography>);
+
+    const dataset = benchmark.dataset({id: datasetId});
+    breadcrumbs.push(<Link to={dataset.home}>{datasetId}</Link>);
+
+    if (!submissionId) {
+      return breadcrumbs;
+    }
+
+    breadcrumbs.push(<Typography>Submissions</Typography>);
+
+    const submission = dataset.submission({id: submissionId});
+    breadcrumbs.push(<Link to={submission.home}>{submissionId}</Link>);
+
+    if (!questionId) {
+      return breadcrumbs;
+    }
+
+    breadcrumbs.push(<Typography>Questions</Typography>);
+
+    breadcrumbs.push(
+      <Link to={submission.question({id: questionId})}>{questionId}</Link>
+    );
+
+    return breadcrumbs;
+  })();
+
+  return <Breadcrumbs>{breadcrumbsChildren}</Breadcrumbs>;
+};
+
+const NotFound: React.FunctionComponent<{label: string}> = ({label}) => {
+  return (
+    <Frame>
+      <Typography variant="h5">{label} was not found</Typography>
+    </Frame>
+  );
+};
+
+const QuestionAnswerChoiceCard: React.FunctionComponent<{
+  choice: QuestionAnswerChoice;
+}> = ({choice, children}) => (
+  <Card>
+    <CardContent>
+      <Grid container>
+        <Grid item xs={2}>
+          <Typography variant="h6">{choice.label}</Typography>
+        </Grid>
+        <Grid item xs={10}>
+          <Typography variant="body1">{choice.text}</Typography>
+        </Grid>
+      </Grid>
+      {children}
+    </CardContent>
+  </Card>
 );
+
+export const BenchmarkAnswerPage: React.FunctionComponent = () => {
+  const variables = _.mapValues(
+    useParams<BenchmarkPathParams>(),
+    decodeURIComponent
+  );
+  const {benchmarkId, datasetId, submissionId, questionId} = variables;
```

Comment:
You could do this decomposition in the line above, since you don't use variables otherwise.